### ISSUE TITLE
Example for `retry.RetryOnConflict`

### ISSUE THREAD

**Issue by dylanhitt** [iximiuz/client-go-examples#5]
The repository should include an example of `retry.RetryOnConflict`.

Current implementation idea:

Folder Name: `retry-on-conflict`
General Flow:
- Create pod -  keep reference
- UpdateStatus pod - ignore reference to emulate the object changing unexpectedly from either another controller or external source
- Implement retry function. To display the use we could use a timer to demonstrate performing long running task to validate the status of an object. For instance sometimes when working with CRDs that create resources external to k8s, gathering status can take a significant amount of time. A very naive solution would be just using a flag and skipping fetch on the first try. Not quite sure, but a simple explanation would be better than a complex example. Maybe someone has some thoughts on how to emulate this situation easily. 
```go
err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
    pod, err := c.Pods("ns").Get(name, metav1.GetOptions{})
    if err != nil {
        return err
    }
    
    // emulate long running tasks to get current status with timers
    
    _, err = c.Pods("ns").UpdateStatus(pod)
})
```
---------------------------------------------------
I'd like to submit a PR for this as I recently hit this position in a custom controller. @iximiuz I'd like to hear thoughts.

**Comment by iximiuz** [comment]
Hi Dylan! The `retry.RetryOnConflict` is one of the must-have examples, so please go for it!

For the actual code, we could probably simplify it a bit. How about:

- Create a ConfigMap
- Read the ConfigMap two times and keep two references
- Update the ConfigMap using ref 1
- Try updating the ConfigMap using ref 2 <-- expect this operation to fail.

Additionally, you could put your explanation for the real-world Pod & controllers use case on the top of this code. This would allow us to keep the mini-program simple (and not run any workloads in the cluster), and wire the example with the real-world situation. WDYT?

**Comment by dylanhitt** [comment]
I understand.

So the actual code execution will fail out completely once the retry limit is reached.

And yeah that sounds good, I think as long as a we have a snippet of it should really look like will suffice.

**Comment by iximiuz** [comment]
Implemented in #6
